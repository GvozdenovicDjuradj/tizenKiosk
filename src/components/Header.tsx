import React from "react"
import { Text, View } from "react-native"
import { KioskTemplate } from "../interfaces"
import { header as headerStyle } from "../theme/styles"

interface Props {
  template?: KioskTemplate
  text: string,
  style?: any

}

export default (props: Props) => {
  const {
      template,
      text,
      style
    } = props
  const templateStyles = {
    header: {},
    text: {},
  }
  if (template) {
    if (template.font) {
      templateStyles.text = {
        ...templateStyles.text,
        fontFamily: template.font
      }
    }
    if (template.headingTextColor) {
      templateStyles.text = {
        ...templateStyles.text,
        color: template.headingTextColor
      }
    }
    if (template.welcomeButtonColor) {
      templateStyles.header = {
        ...templateStyles.header,
        backgroundColor: template.welcomeButtonColor
      }
    }
  }
  return (
    <View style={[headerStyle.view, templateStyles.header, style]}>
      <Text style={[headerStyle.text, templateStyles.text]}>
        {text}
      </Text>
    </View>
  )
}
