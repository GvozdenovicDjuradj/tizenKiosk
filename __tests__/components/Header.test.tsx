import React from "react"
import { shallow } from "enzyme"
import Header from "../../src/components/Header"
import { createTemplate } from "../fixtures"

describe("Header tests", () => {

  it("should render Header successfully applying styles from template provided", () => {
    const template = createTemplate()
    const text = "Header text"
    const wrapper = shallow(
      <Header template={template} text={text} />
    )
    expect(wrapper.prop("style")).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          backgroundColor: template.welcomeButtonColor
        })
      ])
    )
    expect(wrapper.dive()).toMatchSnapshot()
  })

  it("should render Header successfully without template", () => {
    const text = "Header text"
    const wrapper = shallow(
      <Header text={text} />
    )
    expect(wrapper.dive()).toMatchSnapshot()
  })

})
