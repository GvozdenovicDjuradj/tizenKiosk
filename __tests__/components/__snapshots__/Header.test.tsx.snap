// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should render Header successfully applying styles from template provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F80347",
          "justifyContent": "center",
          "marginTop": 80,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#00ff00",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Arial",
            "fontSize": 55,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#fff",
            "fontFamily": "Arial",
          },
        ]
      }
    >
      Header text
    </Text>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
              "fontSize": 55,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
            },
          ]
        }
      >
        Header text
      </Text>,
      "style": Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F80347",
          "justifyContent": "center",
          "marginTop": 80,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#00ff00",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Header text",
        "style": Array [
          Object {
            "color": "#fff",
            "fontFamily": "Arial",
            "fontSize": 55,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#fff",
            "fontFamily": "Arial",
          },
        ],
      },
      "ref": null,
      "rendered": "Header text",
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Arial",
                "fontSize": 55,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "color": "#fff",
                "fontFamily": "Arial",
              },
            ]
          }
        >
          Header text
        </Text>,
        "style": Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F80347",
            "justifyContent": "center",
            "marginTop": 80,
            "padding": 20,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#00ff00",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Header text",
          "style": Array [
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
              "fontSize": 55,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
            },
          ],
        },
        "ref": null,
        "rendered": "Header text",
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
    Symbol(enzyme.__providerValues__): Map {},
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Header tests should render Header successfully without template 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F80347",
          "justifyContent": "center",
          "marginTop": 80,
          "padding": 20,
          "width": "100%",
        },
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Arial",
            "fontSize": 55,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Header text
    </Text>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
              "fontSize": 55,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {},
          ]
        }
      >
        Header text
      </Text>,
      "style": Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F80347",
          "justifyContent": "center",
          "marginTop": 80,
          "padding": 20,
          "width": "100%",
        },
        Object {},
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Header text",
        "style": Array [
          Object {
            "color": "#fff",
            "fontFamily": "Arial",
            "fontSize": 55,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {},
        ],
      },
      "ref": null,
      "rendered": "Header text",
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Arial",
                "fontSize": 55,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {},
            ]
          }
        >
          Header text
        </Text>,
        "style": Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F80347",
            "justifyContent": "center",
            "marginTop": 80,
            "padding": 20,
            "width": "100%",
          },
          Object {},
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Header text",
          "style": Array [
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
              "fontSize": 55,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {},
          ],
        },
        "ref": null,
        "rendered": "Header text",
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
    Symbol(enzyme.__providerValues__): Map {},
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
